### PR TITLE
qa/suites/fs: add prefetch_dirfrags false to thrasher suite

### PR DESCRIPTION
**Pull request by MrFreezeex**
Fixes: https://tracker.ceph.com/issues/52591
Signed-off-by: Arthur Outhenin-Chalandre <arthur.outhenin-chalandre@cern.ch>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests
- Teuthology
  - [ ] Completed teuthology run
  - [ ] No teuthology test necessary (e.g., documentation)

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by MrFreezeex** [comment]
As suggested in the tracker I added the test in the thrasher suite, not sure if it should be a separate test or the new default for the trasher suite (like I'm doing here)...

File: qa/suites/fs/thrash/workloads/overrides/prefetch_dirfrags.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+overrides:
+  ceph:
+    conf:
+      mds:
+        mds oft prefetch dirfrags: false
```
Comment (batrick) [inline qa/suites/fs/thrash/workloads/overrides/prefetch_dirfrags.yaml:5]: Suggest this patch:

````
diff --git a/qa/suites/fs/thrash/overrides/+ b/qa/suites/fs/thrash/overrides/%
similarity index 100%
rename from qa/suites/fs/thrash/overrides/+
rename to qa/suites/fs/thrash/overrides/%
diff --git a/qa/suites/fs/thrash/overrides/prefetch_dirfrags/no.yaml b/qa/suites/fs/thrash/overrides/prefetch_dirfrags/no.yaml
new file mode 100644
index 00000000000..91b45367934
--- /dev/null
+++ b/qa/suites/fs/thrash/overrides/prefetch_dirfrags/no.yaml
@@ -0,0 +1,5 @@
+overrides:
+  ceph:
+    conf:
+      mds:
+        mds oft prefetch dirfrags: false
diff --git a/qa/suites/fs/thrash/overrides/prefetch_dirfrags/yes.yaml b/qa/suites/fs/thrash/overrides/prefetch_dirfrags/yes.yaml
new file mode 100644
index 00000000000..bd202f988c8
--- /dev/null
+++ b/qa/suites/fs/thrash/overrides/prefetch_dirfrags/yes.yaml
@@ -0,0 +1,5 @@
+overrides:
+  ceph:
+    conf:
+      mds:
+        mds oft prefetch dirfrags: true
````

So we test both configurations.

**Review by batrick — CHANGES_REQUESTED** [review]
We need to test both configurations, suggest structuring like:

````
.../overrides/prefetch_dirfrags/yes.yaml
.../overrides/prefetch_dirfrags/no.yaml
````

File: qa/suites/fs/thrash/workloads/overrides/prefetch_dirfrags.yaml (hunk 1 above)
Comment (MrFreezeex) [reply qa/suites/fs/thrash/workloads/overrides/prefetch_dirfrags.yaml]: I applied it, thanks!

**Comment by vshankar** [comment]
@MrFreezeex I think the convolution operator `qa/suites/fs/thrash/workloads/overrides/prefetch_dirfrags/%` should be a directory level up. @batrick ?

**Comment by batrick** [comment]
> @MrFreezeex I think the convolution operator `qa/suites/fs/thrash/workloads/overrides/prefetch_dirfrags/%` should be a directory level up. @batrick ?

Yes, and the `+` operator needs deleted.

**Comment by MrFreezeex** [comment]
> Yes, and the `+` operator needs deleted.

Ah I didn't applied your patch correctly indeed sorry. Just to understand this would apply the convulution operator on all the other override as well and thus add much more tests than just the dirfrags things right?

**Comment by batrick** [comment]
> > Yes, and the `+` operator needs deleted.
> 
> Ah I didn't applied your patch correctly indeed sorry. Just to understand this would apply the convolution operator on all the other overrides as well and thus add much more tests than just the dirfrags one right?

No. The other overrides will still be combined as if the `+` operator will be used. The `%` operator only affects the `prefetch_dirfrags` directory.

**Review by batrick — APPROVED** [review]

**Review by vshankar — APPROVED** [review]

**Comment by vshankar** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Master#2022-Jan-03

**Pull request merged**